### DISCUSSION TITLE
Make it easier to see the currently thrown exception

### DISCUSSION THREAD

**Discussion by bhackett1024** [RecordReplay/devtools#6114] [Ideas]
When using replay I often find myself interested in the exceptions that have been thrown.  It's hard to see these thrown exceptions except by turning on the "exceptions" analysis in the console, which will only work if there are relatively few exceptions that were thrown across the entire recording.  This also makes it hard to refer to exceptions in comments, as someone else viewing the recording won't see the console message for that exception.

I think it would be nice to show the current exception in the pause pane when we're paused at a place where an exception is available --- we're at a site associated with an exception throw or a frame unwind.  We have Pause.getExceptionValue for this, but it will only work at the exception throw and not at unwinds, though we could update the backend to support that use case.

Ideally, when in a frame that throws an exception there should be an easy way to jump to that exception site, either with the step out button (though that normally goes to the calling frame...) or by scrubbing to the last point in the frame scrubber.

**Comment by gideonred** [comment]
Just a random thought: What about a red triangle next to each line of code in the code editor at which an exception was thrown? That might help someone quickly identify where exceptions are happening.

**Comment by jasonLaster** [comment]
> I think it would be nice to show the current exception in the pause pane when we're paused at a place where an exception is available --- we're at a site associated with an exception throw or a frame unwind. We have Pause.getExceptionValue for this, but it will only work at the exception throw and not at unwinds, though we could update the backend to support that use case.

Agreed.

Can you explain what an unwind is?

---

There's some prior-art here as well in terms of explaining more pause context: paused at breakpoint, paused on step, paused on dom mutation. We could do something similar in this context. Here's a quick design based on what others do here. Curious what others think

https://www.figma.com/file/SSr1ljyzF0lCXTL7BX1Whh/Replay-Design-Doc?node-id=10575%3A107316
<img width="1624" alt="Frame 2 (4)" src="https://user-images.githubusercontent.com/254562/157562823-49ba0076-1d0c-4343-b57b-380cac7c865d.png">

**Comment by bhackett1024** [comment]
An unwind is the point at which a frame exits due to an exception being thrown.

**Comment by jaril** [comment]
@jonbell-lot23 can you put this on a queue for us to discuss?

**Comment by jonbell-lot23** [comment]
**Update**

Jaril and I had a chat and thought this was a good candidate for a discussion. Not because we're de-prioritising, but literally because we're wanting to open some more discussion about it.

**Design POV**

I quite like both Jason and Gideon's ideas -- a little message at the top left in Pause Info as well as a little icon on the line number itself in the editor. Both have precedent and are pretty straight-forward and helpful for users.

**Discussion points**

Jaril was thinking through the use cases that gets someone to this state, and realised this is perhaps a different flow than he'd normally follow, so he wanted to think about and discuss it more to gain context.

**Comment by jasonLaster** [comment]
![image](https://user-images.githubusercontent.com/254562/161165924-8500c72c-ac14-4338-860b-6483ab8bad53.png)
Here's the pause badge in action
https://developer.chrome.com/docs/devtools/javascript/
